Library/OcConsoleLib/TextOutputBuiltin.c

```diff
@@ -919,8 +948,10 @@ ConsoleControlInstall (
 }
 
 EFI_STATUS
-OcUseBuiltinTextOutput (
```

Comment:
I do not really think we need two functions, just call AsciiTextResetEx with TRUE, TRUE in any case. Also, since it is basically always called with TRUE+TRUE or FALSE+FALSE the argument could be one, e.g. Debug.

Reply:
Sorry, a bit lost on that.

In AsciiTextReset we call:
`Status = AsciiTextResetEx (This, ExtendedVerification, false, FALSE);`

OcUseBuiltinTextOutputEx, when called from OcUseBuiltinTextOutput, calls:
`Status = AsciiTextResetEx (&mAsciiTextOutputProtocol, true, true, FALSE);`

The last, `RawStatus`, param is always `FALSE` on the normal internal calls using `AsciiTextReset` and `OcUseBuiltinTextOutput` to keep everything working as you wanted (logs the real status on error but returns `EFI_DEVICE_ERROR`). 

This works fine for doing everything in OpenCore but needed a bit more.

OcUseBuiltinTextOutputEx, when called directly, can call:
`Status = AsciiTextResetEx (&mAsciiTextOutputProtocol, true, true, TRUE);`

The `RawStatus` param can be set to `TRUE` to have the raw/real status returned instead of `EFI_DEVICE_ERROR`.

Not sure how this can be achieved in another way without adding an additional param to `OcUseBuiltinTextOutput` and editing all instances where it is called.

Reply:
> Not sure how this can be achieved in another way without adding an additional param to OcUseBuiltinTextOutput and editing all instances where it is called.

Actually the complication is the optional verbose logging. If you are happy for calls to `AsciiTextResetEx` from both `AsciiTextReset` and `OcUseBuiltinTextOutput` to have verbose logging, then this end can be achieved without an additional function.

Reply:
It does not matter what the AsciiTextResetEx and OcUseBuiltinTextOutput functions return, since they are for private use. AsciiTextReset on the other side must adhere to the protocol interface AND it cannot print due to a possibility of a recursion. For this reason I suggested to:

* drop OcUseBuiltinTextOutputEx and edit OcUseBuiltinTextOutput instead making it behave as OcUseBuiltinTextOutputEx(Mode, TRUE, TRUE) currently.
* edit AsciiTextResetEx to take only one extra parameter (Debug) that will enable both verbose logging and raw error code.
* edit OcUseBuiltinTextOutput to call AsciiTextResetEx with Debug = TRUE and let the other functions call it with FALSE.

This looks best to me, sorry for being unclear earlier.